### DISCUSSION TITLE
# Proposal: Migrate Profiles NFT to Darwinia

### DISCUSSION THREAD

**Discussion by furoxr** [ringdao-community-guild/discussion#30] [General]
## Introduction

This proposal propose the migration of Profiles NFT to Darwinia. After months of work of DCDAO, we have made integration with Snapshot and have [self-hosted Safe Wallet](https://multisig.darwiniacommunitydao.xyz/). And the conditions are already fulfilled, we could migrate the NFT contract to Darwinia, which won’t impact the operation of DCDAO. What’s more, we could benefit from the migration.

## Background

Profiles NFT is the membership of DCDAO, and it works as the core of operation of DCDAO. Every NFT owner possesses the right to propose ideas and influence DCDAO development through voting privileges. 

At present, we are using Snapshot to manage proposal, and we have a basic strategy for voting: every NFT counts as one vote, proposals passing majority and exceeding minimal participation threshold are considered approved.

We also have a multisig wallet controled by five member for now, and it will handle the fund related to the proposal. During the initial founding of DCDAO, both Snapshot and Safe did not support Darwinia, so choices limited to only Profile NFTs being created on Ethereum back then. 

And we have [self-hosted Safe Wallet](https://multisig.darwiniacommunitydao.xyz/) and support of Snapshot, I think it’s time to migrate our Profiles NFT contract to Darwinia. As Darwinia Community DAO, fostering and expanding the Darwinia ecosystem is critical, besides it will notably reduce the costs of minting NFT to new members. 

## Migration procedure

1. Create a multisig wallet at Darwinia with the same member. The detailed members are in this [proposal](https://snapshot.org/#/darwiniacommunitydao.eth/proposal/0x045f364dcbc7c1d41d51d78fe85473e4a5aa49018aecc2b3bee0d97496d04921)

2. Deploy [Profiles NFT](https://github.com/dcdao/profiles) contract to Darwinia

3. Mint NFTs to current members up to the time this proposal adopted

4. Change Snapshot strategies to use Profiles NFT on Darwinia

After the proposal adopted, the entire migration work should be done in a week. And some tests are necessary to ensure the success of migration.

**Comment by 2xJDubs** [comment]
Think this is a great idea as the NFTs determining membership should live on Darwinia.  Understand that initially this wasn't possible so definitely time to migrate.  This also shows other possible DAO's that they can use Darwinia.

**Comment by furoxr** [comment]
Migration completed~ I'll update strategies on Snapshot later:

Multisig Wallet on Darwinia: 0x8683A0ED2bAdb1F3AaBeA6686d78649F0Da774A0
DCDAO NFT Address on : 0x02bE69cF9172a3DcbB334DB9DE41E899F4A8481D

And this is the transaction of migration: https://darwinia.subscan.io/tx/0xa6ee6192ceae00a2984a17c87778e2e8f6e41c92012d3207e75df8b1a15cafca
